### DISCUSSION TITLE
Best way to identify Canon and Nikon RAW tiff files

### DISCUSSION THREAD

**Discussion by hanssonrickard** [libvips/libvips#3898] [Q&A]
Hi.
We are using pyvips to process images, sometimes we get tiff images that are RAW files from Canon or Nikon cameras and the result is them whacko.
Is there any way to easy identify those images when they load in pyvips, via some kind of tag or info and then decide to process those images using rawpy instead of pyvips (and let pyvips deal with "real" tiffs only)?

**Comment by jcupitt** [comment]
Hello @hanssonrickard,

You could look for files with the nikon tags (0x9003 and 0x9216). You'd need some other lib to read these tags, maybe `tifffile`?

exiftool is super easy, though probably a bit slower:

```
$ exiftool image1.NEF -MIMEType
MIME Type                       : image/x-nikon-nef
```

**Reply by jcupitt** [reply]
libvips passes files like NEF on to imagemagick, which in turn usually uses `ufraw-batch`. Under the hood, that ought to use the same loader and therefore give the same result as rawpy, I think.

Relying on imagemagick has its own set of issues, of course.

**Comment by hanssonrickard** [comment]
Thanks.
I will consider if i should build libvips again, but with imagemagick support as well (it is not built with thhat at the moment), or if i will go with the exiftool check instead and use rawpy instead.
